spec/prog/rotate_storage_kek_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require_relative "../model/spec_helper"
+
+RSpec.describe Prog::RotateStorageKek do
+  subject(:rsk) {
+    described_class.new(Strand.new(prog: "RotateStorageKek",
+      stack: [{vm_storage_volume_id: "bogus"}]))
+  }
+
+  let(:sshable) {
+    instance_double(Sshable)
+  }
+
+  let(:vm) {
+    vm_host = instance_double(VmHost)
+    vm = Vm.new(size: "m5a.2x").tap {
+      _1.id = Vm.generate_uuid
+    }
+    allow(vm_host).to receive(:sshable).and_return(sshable)
+    allow(vm).to receive(:vm_host).and_return(vm_host)
+    vm
+  }
+
+  let(:current_kek) {
+    StorageKeyEncryptionKey.new(
+      algorithm: "aes-256-gcm", key: "key_1",
+      init_vector: "iv_1", auth_data: "somedata"
+    ) { _1.id = StorageKeyEncryptionKey.generate_uuid }
+  }
+
+  let(:new_kek) {
+    StorageKeyEncryptionKey.new(
+      algorithm: "aes-256-gcm", key: "key_2",
+      init_vector: "iv_2", auth_data: "somedata"
+    ) { _1.id = StorageKeyEncryptionKey.generate_uuid }
+  }
+
+  let(:volume) {
+    disk = VmStorageVolume.new(boot: true, size_gib: 20, disk_index: 0)
+    disk.key_encryption_key_1 = current_kek
+    disk.key_encryption_key_2 = new_kek
+    disk.vm = vm
+    disk
+  }
+
+  before do
+    allow(rsk).to receive(:vm_storage_volume).and_return(volume)
+  end
+
+  describe "#start" do
+    it "creates a key & hops to install" do
+      expect(StorageKeyEncryptionKey).to receive(:create).and_return(current_kek)
+      expect(volume).to receive(:update).with({key_encryption_key_2_id: current_kek.id})
+      expect(rsk).to receive(:hop).with(:install)
+      rsk.start
```

Comment:
We have `hop` rspec matcher, may be you can use it
```suggestion
      expect { rsk.start }.to hop("install")
```